examples/broadcasting_your_own_methods.py

```diff
@@ -0,0 +1,472 @@
+positions: xr.DataArray = ds.position
+
+# %%
+# The individuals in this dataset follow very similar, arc-like trajectories.
+# To help emphasise what we are doing in this example, we will offset the paths
+# of two of the individuals by a small amount so that the trajectories are more
+# distinct.
+
+positions.loc[:, "y", :, "AEON3B_TP1"] -= 100.0
+positions.loc[:, "y", :, "AEON3B_TP2"] += 100.0
+
+fig, ax = plt.subplots(1, 1)
+for mouse_name, col in zip(
+    positions.individuals.values, ["r", "g", "b"], strict=False
+):
+    ax.plot(
+        positions.sel(individuals=mouse_name, space="x"),
+        positions.sel(individuals=mouse_name, space="y"),
+        linestyle="-",
+        marker=".",
+        markersize=2,
+        linewidth=0.5,
+        c=col,
+        label=mouse_name,
+    )
+    ax.invert_yaxis()
+    ax.set_xlabel("x (pixels)")
+    ax.set_ylabel("y (pixels)")
+    ax.axis("equal")
+    ax.legend()
+
+# %%
+# Motivation
+# ----------
+# Suppose that, during our experiment, we have a region of the enclosure that
+# has a slightly wet floor, making it slippery. The individuals must cross this
+# region in order to reach some kind of reward on the other side of the
+# enclosure.
+# We know that the "slippery region" of our enclosure is approximately
+# rectangular in shape, and has its opposite corners at (400, 0) and
+# (600, 2000), where the coordinates are given in pixels.
+# We could then write a function that determined if a given (x, y) position was
+# inside this "slippery region".
+
+
+def in_slippery_region(xy_position) -> bool:
+    """Return True if xy_position is in the slippery region.
+
+    Return False otherwise.
+    xy_position has 2 elements, the (x, y) coordinates respectively.
+    """
+    slippery_region_bl_corner = (400.0, 0.0)
+    slippery_region_tr_corner = (600.0, 2000.0)
+
+    is_within_bounds_x = (
+        slippery_region_bl_corner[0]
+        <= xy_position[0]
+        <= slippery_region_tr_corner[0]
+    )
+    is_within_bounds_y = (
+        slippery_region_bl_corner[1]
+        < xy_position[1]
+        <= slippery_region_tr_corner[1]
+    )
+    return is_within_bounds_x and is_within_bounds_y
+
+
+# We can just check our function with a few sample points
+for point in [(0, 100), (450, 700), (550, 1500), (601, 500)]:
+    print(f"{point} is in slippery region: {in_slippery_region(point)}")
+
+# %%
+# Determine if each position can be seen
+# --------------------------------------
+# Given our data, we could extract whether each position (for each time-point,
+# and each individual) was inside the slippery region by looping over the
+# values.
+
+data_shape = positions.shape
+in_view = np.zeros(
+    shape=(
+        len(positions["time"]),
+        len(positions["keypoints"]),
+        len(positions["individuals"]),
+    ),
+    dtype=bool,
+)  # We would save one result per time-point, per key-point, per individual
```

Comment:
```suggestion
)  # We would save one result per time-point, per keypoint, per individual
```
We use "keypoint" (without hyphen) throughout `movement`.